### PR TITLE
site: documentation for TLS client authentication

### PR DESCRIPTION
**Pull request by tsaarni**
Added documentation describing the external client certificate validation.

Signed-off-by: Tero Saarni <tero.saarni@est.tech>

### CODE REVIEW

**Comment by tsaarni** [comment]
Fixes #2349

I did this PR on top of Contour 1.3 documentation to get early feedback.  However, this should NOT be merged as such, since the documentation should land in release which will have the API parts of 
#2250.

File: site/docs/v1.3.0/httpproxy.md [hunk 1]
```diff
@@ -1263,6 +1263,35 @@ spec:
             subjectName: foo.marketing
 ```
 
+## Validation of External Client Certificate
```
Comment (jpeach) [inline site/docs/v1.3.0/httpproxy.md]: Lets title this "Client Certificate Validation"?

File: site/docs/v1.3.0/httpproxy.md [hunk 2]
```diff
@@ -1263,6 +1263,35 @@ spec:
...
+
+It is possible to protect the backend service from unauthorized external clients by requiring the client to present a valid TLS certificate.
+Envoy will validate the client certificate by verifying that a chain of trust can be established from the presented client certificate to the configured trusted root CA certificate.
+Also, all the involved certificates must be unexpired.
```
Comment (jpeach) [inline site/docs/v1.3.0/httpproxy.md]: I think it scans a bit better if we combine these sentences:
```
Envoy will validate the client certificate by verifying that it is not expired and that a a chain of trust can be established to the configured trusted root CA certificate.
```

WDYT?

File: site/docs/v1.3.0/httpproxy.md [hunk 3]
```diff
@@ -1263,6 +1263,35 @@ spec:
...
+      secretName: secret
+      clientValidation:
+        caSecret: client-root-ca
+  routes:
+    - services:
+        - name: s1
+          port: 80
+```
+
+The preceding example enables validation by defining optional `clientValidation` attribute.
```
Comment (jpeach) [inline site/docs/v1.3.0/httpproxy.md]: s/by defining/by setting the/

File: site/docs/v1.3.0/httpproxy.md (hunk 1 above)
Comment (tsaarni) [reply site/docs/v1.3.0/httpproxy.md]: done!

File: site/docs/v1.3.0/httpproxy.md (hunk 2 above)
Comment (tsaarni) [reply site/docs/v1.3.0/httpproxy.md]: Sounds better to me too.

File: site/docs/v1.3.0/httpproxy.md (hunk 3 above)
Comment (tsaarni) [reply site/docs/v1.3.0/httpproxy.md]: done!

**Comment by tsaarni** [comment]
Thanks for reviewing!  Sorry, somehow I managed to miss that that there is "site/docs/master/" for development of docs.  I've now moved my change there :)

File: site/docs/master/httpproxy.md [hunk 4]
```diff
@@ -1263,6 +1263,34 @@ spec:
...
+      clientValidation:
+        caSecret: client-root-ca
+  routes:
+    - services:
+        - name: s1
+          port: 80
+```
+
+The preceding example enables validation by setting the optional `clientValidation` attribute.
+Its mandatory attribute `caSecret` contains a name of existing Kubernetes Secret that must be of type "Opaque" and have a data key named as `ca.crt`.
```
Comment (jpeach) [inline site/docs/master/httpproxy.md:1291]:
```suggestion
Its mandatory attribute `caSecret` contains a name of an existing Kubernetes Secret that must be of type "Opaque" and have a data key named `ca.crt`.
```

Comment (tsaarni) [reply site/docs/master/httpproxy.md]: Thanks, fixed!  I appreciate having native speaker go through the text :)

**Pull request merged**